### PR TITLE
Socket gateway for notification system

### LINKED ISSUE #231: Implement socket gateway for notification system

### PR DESCRIPTION
**Pull request by b3h3m0th**
| Q                    | A                                                  |
| -------------------- | -------------------------------------------------- |
| Bug fix?             | no |
| New feature?         | yes                                             |
| Related issues/PRs   | closes #231 |
| External ticket link |      |
| License              | MIT                                                |

### CODE REVIEW

File: libs/client/common/socket/src/lib/socket.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { User } from '@bregenz-bewegt/client/types';
+import {
+  ClientToServerEvents,
+  ServerToClientEvents,
+} from '@bregenz-bewegt/shared/types';
+import { io, ManagerOptions, Socket, SocketOptions } from 'socket.io-client';
+
+const connectionOptions: Partial<ManagerOptions & SocketOptions> = {
+  forceNew: true,
+  reconnectionAttempts: Infinity,
```
Comment (TimonStadelmann) [inline libs/client/common/socket/src/lib/socket.ts:10]: bitte kleiner

File: libs/client/ui/components/src/lib/notification-listener/notification-listener.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        icon: notificationIcon,
+        duration: 2000,
+        position: 'top',
+        mode: 'ios',
+        color: 'light',
+        buttons: [{ icon: close, role: 'cancel' }],
+      });
+    };
+
+    const handleFetchNotification = () => {
```
Comment (TimonStadelmann) [inline libs/client/ui/components/src/lib/notification-listener/notification-listener.tsx:44]: funktion brucht es net

File: libs/client/ui/pages/src/lib/notifications/notifications.tsx [hunk 3]
```diff
@@ -34,16 +36,20 @@ import { checkmarkDone, trash } from 'ionicons/icons';
...
+  notificationsStore.storeKey,
+  userStore.storeKey
 )(
-  observer(({ notificationsStore }) => {
+  observer(({ notificationsStore, userStore }) => {
     const [presentDefaultErrorToast] = useDefaultErrorToast();
     const [isGuest] = useIsGuest();
 
     const deleteNotification = (notificationId: Notification['id']) => {
+      if (isGuest) return;
```
Comment (TimonStadelmann) [inline libs/client/ui/pages/src/lib/notifications/notifications.tsx:51]: Gäste sollen o notifications kriga künna

File: libs/server/controllers/friend/src/lib/friend.service.ts [hunk 4]
```diff
@@ -79,19 +83,17 @@ export class FriendService {
...
       data: {
-        notifications: {
-          create: {
-            title: 'Neue Freundschaftsanfrage',
-            description: `${requestee.username} hat dir eine Freundschaftsanfrage gesendet`,
-            type: NotificationType.FRIEND_REQUEST_RECEIVED,
-          },
-        },
+        title: 'Neue Freundschaftsanfrage',
+        description: `${requestee.username} hat dir eine Freundschaftsanfrage gesendet`,
```
Comment (TimonStadelmann) [inline libs/server/controllers/friend/src/lib/friend.service.ts:89]: Line Breaks in da Description

**Review by TimonStadelmann — CHANGES_REQUESTED** [review]

File: libs/client/ui/pages/src/lib/notifications/notifications.tsx (hunk 3 above)
Comment (b3h3m0th) [reply libs/client/ui/pages/src/lib/notifications/notifications.tsx:51]: #238

File: libs/client/ui/components/src/lib/notification-listener/notification-listener.tsx (hunk 2 above)
Comment (b3h3m0th) [reply libs/client/ui/components/src/lib/notification-listener/notification-listener.tsx:44]: joa isch halt fürd readability

File: libs/server/controllers/friend/src/lib/friend.service.ts (hunk 4 above)
Comment (b3h3m0th) [reply libs/server/controllers/friend/src/lib/friend.service.ts:89]: Beim toast passts glob e:
image

Mit breaks würds so usluaga:
image

**Review by TimonStadelmann — APPROVED** [review]

File: libs/server/controllers/friend/src/lib/friend.service.ts (hunk 4 above)
Comment (TimonStadelmann) [reply libs/server/controllers/friend/src/lib/friend.service.ts:89]: sehr nice👍, isch es aktuell so?

Comment (b3h3m0th) [reply libs/server/controllers/friend/src/lib/friend.service.ts:89]: yes

**Pull request merged**